cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -400,6 +404,18 @@ public void initialize(MasterEnvironmentContext context) throws IOException, Ill
     LOG.info("Kubernetes environment initialized with pod labels {}", podLabels);
   }
 
+  private void validateClusterRole(String roleName) throws ApiException, IllegalArgumentException {
+    try {
+      rbacV1Api.readClusterRole(roleName, null);
+    } catch (ApiException e) {
+      if (e.getCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+        throw new IllegalArgumentException(String.format("ClusterRole '%s' does not exist. Ensure it is created and " +
+                                                           "assigned to the master service account.", roleName), e);
+      }
+      throw e;
```

Comment:
can you wrap this in another exception in order to include `e.getResponseBody()`? IIRC the message in ApiException is useless for debugging

Reply:
Fixed